rundeckapp/src/test/groovy/rundeck/services/ScheduledExecutionServiceSpec.groovy

```diff
@@ -3408,21 +3431,27 @@ class ScheduledExecutionServiceSpec extends Specification implements ServiceUnit
         def result = service.getJobIdent(job, exec)
 
         then:
-        result.jobname == jobname.replaceAll('_ID_', "$id").replaceAll('_JID_', "${job?.id}")
-        result.groupname == groupname.replaceAll('_ID_', "$id").replaceAll('_JID_', "${job?.id}")
+        if(!isJob && !jobScheduled && !executionType.equals("scheduled")){
+            result.jobname == jobname.replaceAll('_ID_', "$id").replaceAll('_JID_', "${job?.id}")
+            result.groupname == groupname.replaceAll('_ID_', "$id").replaceAll('_JID_', "${job?.id}")
+        }else{
+            result.jobname == jobname
+            result.groupname == groupname
```

Comment:
These tests will all pass even if the statements return false, because they must be asserted!

This is a pitfall of using `if` statements inside the Spock then/expect sections!

```suggestion
            assert result.jobname == jobname.replaceAll('_ID_', "$id").replaceAll('_JID_', "${job?.id}")
            assert result.groupname == groupname.replaceAll('_ID_', "$id").replaceAll('_JID_', "${job?.id}")
        }else{
            assert result.jobname == jobname
            assert result.groupname == groupname
```